responsibleai/model_monitor/components/src/calculate_attribution.py

```diff
@@ -0,0 +1,177 @@
+
+    Args:
+      task_type: str, The task type (regression or classification) of the resulting model
+      target_column: str, the column to predict
+      baseline_df: The baseline dataframe
+      production_df: The production dataframe
+
+    Returns:
+       An appropriate model wrapper
+    """
+    y_train = baseline_df[target_column]
+    x_train = baseline_df.drop([target_column], axis=1)
+    x_test = production_df.drop([target_column], axis=1)
+    model = create_lightgbm_model(x_train, y_train, task_type)
+
+    if task_type == 'classification':
+        all_data = pd.concat([x_test, x_train])
+        model_predict = model.predict(all_data)
+        model_predict_proba = model.predict_proba(all_data)
+        model_wrapper = PredictionsModelWrapperClassification(
+            all_data,
+            model_predict,
+            model_predict_proba)
+    else:
+        all_data = pd.concat([x_test, x_train])
+        model_predict = model.predict(all_data)
+        model_wrapper = PredictionsModelWrapperRegression(all_data, model_predict)
+
+    _logger.info("Created ml wrapper")
+    return model_wrapper
+
+
+def compute_categorical_features(baseline_df, target_column):
+    """Compute which features are categorical based on data type of the columns.
+
+    Args:
+      baseline_df: The baseline dataframe
+      target_column: str, the column to predict
+
+    Returns:
+      list: A list of categorical features.
+    """
+    categorical_features = []
+    for column in baseline_df.columns:
+        baseline_column = pd.Series(baseline_df[column])
+        if baseline_column.dtype.name == 'object' and baseline_column.name != target_column:
+            categorical_features.append(baseline_column.name)
+    _logger.info("Successfully categorized columns")
+    return categorical_features
+
+
+def compute_explanations(model_wrapper, df, categorical_features, target_column, task_type):
+    """Compute explanations (feature importances) for a given dataset
+
+    Args:
+      model_wrapper: wrapper around a model that can be used to calculate explanations
+      df: The dataframe
+      categorical_features: list of categorical features not including the target column
+      target_column: str, the column to predict
+      task_type: str, the task type (regression or classification) of the resulting model
+
+    Returns:
+      list: A list of explanations.
+    """
+    # Create the RAI Insights object, use either baseline as train and test data
+    rai_i: RAIInsights = RAIInsights(
+        model_wrapper, df, df, target_column, task_type, categorical_features=categorical_features
+    )
+
+    # Add the explanation and compute
+    rai_i.explainer.add()
+    rai_i.compute()
+
+    return rai_i.explainer._explanation.global_importance_values
+
+
+def calculate_attribution_drift(baseline_explanations, production_explanations):
+    """Compute feature attribution drift given two sets of explanations
+
+    Args:
```

Comment:
Great comments in code.  Interesting how this style differs from most of the other AzureML repos.